router/queryrange.go

```diff
@@ -119,30 +125,46 @@ func parseLegacyRequest(req *pbsearch.RouterRequest, head uint64, lib uint64, he
 
 	if req.Descending {
 		if req.StartBlock == 0 {
-			// FIXME: why do we add StartBLock when it should be 0
-			highBlkNum = virtualHead + int64(req.StartBlock)
-			if highBlkNum < 1 {
-				return nil, derr.Statusf(codes.InvalidArgument, "invalid start block: head or lib - %d goes beyond first block (head: %d, lib: %d)", -highBlkNum, head, lib)
+			highBlkNum = virtualHead
+			if highBlkNum < int64(absoluteTruncationLowBlockNum) {
+				return nil, derr.Statusf(codes.InvalidArgument, "invalid start block: (head or lib - %d) is lower than the lowest block served by this endpoint [%d] (head: %d, lib: %d)", req.StartBlock, absoluteTruncationLowBlockNum, head, lib)
```

Comment:
`lib - %d` will always yield `lib - 0` since we use `req.StartBlock` which must be 0 in this `if` branch.